sktime/markov/msm/augmented_msm.py

```diff
@@ -0,0 +1,623 @@
+            Symmetrized count matrix :math:`C_\mathrm{sym} = 0.5 (C + C^\top)` .
+        count_matrix_row_sums : (N, ) ndarray
+            Row sums of count matrix.
+        """
+        self.lagrange = np.zeros_like(experimental_measurements)
+        self.pi = stationary_distribution
+        self.pi_hat = np.copy(stationary_distribution)
+        self.m_hat = None
+        self.experimental_measurements = experimental_measurements
+        self.experimental_measurement_weights = experimental_measurement_weights
+        self.expectations_by_state = expectations_by_state
+        self.tensor_cache_size = tensor_cache_size
+        self.symmetrized_count_matrix = symmetrized_count_matrix
+        self.count_matrix_row_sums = count_matrix_row_sums
+        self.X = np.empty_like(symmetrized_count_matrix)
+
+        self.m_hat = np.empty(())
+        self._slope_obs = None
+        self.update_m_hat()
+        self.d_m_hat = 1e-1 * np.ones_like(self.m_hat)
+        self.R_slices = np.empty(())
+        self.R_slices_i = 0
+        self.update_R_slices(0)
+        self.Q = np.zeros((self.n_states, self.n_states), dtype=expectations_by_state.dtype)
+        self.G = np.empty(())
+        self._log_likelihood_prev = None
+        self.log_likelihoods = []
+
+    @property
+    def log_likelihood_prev(self):
+        r""" Property storing the previous log likelihood. Used in optimization procedure.
+
+        :getter: gets the previous log likelihood
+        :setter: sets the previous log likelihood
+        :type: float
+        """
+        return self._log_likelihood_prev
+
+    @log_likelihood_prev.setter
+    def log_likelihood_prev(self, value):
+        self._log_likelihood_prev = value
+
+    @property
+    def n_states(self):
+        r""" Number of discrete states (N) that fall into the considered state set. """
+        return self.expectations_by_state.shape[0]
+
+    @property
+    def n_experimental_observables(self):
+        r""" Number of experimental observables (K). """
+        return self.expectations_by_state.shape[1]
+
+    @property
+    def slope_obs(self) -> np.ndarray:
+        r""" Numerically computed slope for Newton algorithm. """
+        return self._slope_obs
+
+    def update_m_hat(self):
+        r""" Updates m_hat (expectation of observable of the Augmented Markov model) """
+        self.m_hat = self.pi_hat.dot(self.expectations_by_state)
+        self._slope_obs = self.m_hat - self.experimental_measurements
+
+    def update_R_slices(self, i):
+        """ Computation of multiple slices of R tensor.
+
+        When estimate(.) is called the R-tensor is split into segments whose maximum size is
+        specified by max_cache argument (see constructor).
+        R_slices_i specifies which of the segments are currently in cache. For equations check SI of [1].
+
+        """
+        pek = self.pi_hat[:, None] * self.expectations_by_state[:, i * self.tensor_cache_size:(i + 1) * self.tensor_cache_size]
```

Comment:
`pek` and `pp` for me tough as well - could add something very brief to the method docs?

Reply:
`pek` and `pp` are temporaries so I dont think they should go into the docstring really. They reflect what is presented in SI

Reply:
if they're in the SI then cool